### PR TITLE
Included area of n sided regular polygon

### PR DESCRIPTION
**Pull request by mislah**
Added a function to calculate the area of n sided regular polygons

### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/area.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 """
 Find the area of various geometric shapes
 """
```
Comment (ZeroDayOwl) [inline maths/area.py:3]: Add a credible source as a reference.

File: maths/area.py [hunk 2]
```diff
@@ -458,6 +458,49 @@ def area_rhombus(diagonal_1: float, diagonal_2: float) -> float:
     return 1 / 2 * diagonal_1 * diagonal_2
```
Comment (ZeroDayOwl) [inline maths/area.py:458]: Perhaps add some doctests here too.

File: maths/area.py [hunk 3]
```diff
@@ -458,6 +458,49 @@ def area_rhombus(diagonal_1: float, diagonal_2: float) -> float:
...
+    elif length < 0:
+        raise ValueError(
+            "area_reg_polygon() only accepts non-negative values as \
+length of a side"
+        )
+    return (sides * length**2) / (4 * tan(pi / sides))
+
+
 if __name__ == "__main__":
     import doctest
```
Comment (ZeroDayOwl) [inline maths/area.py:505]:
```suggestion
if __name__ == "__main__":
    import doctest
    doctest.testmod()
```

File: maths/area.py [hunk 4]
```diff
@@ -481,3 +524,6 @@ def area_rhombus(diagonal_1: float, diagonal_2: float) -> float:
     print(f"Cone: {surface_area_cone(10, 20) = }")
     print(f"Conical Frustum: {surface_area_conical_frustum(10, 20, 30) = }")
     print(f"Cylinder: {surface_area_cylinder(10, 20) = }")
+    print(f"Equilateral Triangle: {area_reg_polygon(3, 10) = }")
+    print(f"Square: {area_reg_polygon(4, 10) = }")
+    print(f"Reqular Pentagon: {area_reg_polygon(5, 10) = }")
```
Comment (ZeroDayOwl) [inline maths/area.py:529]: You don't need them since you already used doctest to test your code.

**Review by ZeroDayOwl — CHANGES_REQUESTED** [review]

File: maths/area.py [hunk 5]
```diff
@@ -458,6 +458,49 @@ def area_rhombus(diagonal_1: float, diagonal_2: float) -> float:
...
+        raise ValueError(
+            "area_reg_polygon() only accepts integers greater than or \
+equal to three as number of sides"
+        )
+    elif length < 0:
+        raise ValueError(
+            "area_reg_polygon() only accepts non-negative values as \
+length of a side"
+        )
+    return (sides * length**2) / (4 * tan(pi / sides))
```
Comment (CaedenPH) [inline maths/area.py:501]:
```suggestion
    return (sides * pow(length, 2)) / (4 * tan(pi / sides))
```

File: maths/area.py [hunk 6]
```diff
@@ -458,6 +458,49 @@ def area_rhombus(diagonal_1: float, diagonal_2: float) -> float:
     return 1 / 2 * diagonal_1 * diagonal_2
 
 
+def area_reg_polygon(sides: int, length: float) -> float:
+    """
+    Calculate the area of a regular polygon.
+    Wikipedia reference: https://en.wikipedia.org/wiki/Polygon#Regular_polygons
+    Formula: (n*s^2*cot(pi/n))/4
+    >>> area_reg_polygon(3, 10)
```
Comment (CaedenPH) [inline maths/area.py:466]:
```suggestion

    >>> area_reg_polygon(3, 10)
```

Comment (mislah) [reply maths/area.py:466]: Thank you @CaedenPH for your suggestion.

File: maths/area.py (hunk 5 above)
Comment (mislah) [reply maths/area.py:501]: Hi @CaedenPH , Based on the format of preexisting functions, I doubt if I'm allowed to commit your suggestion.

File: maths/area.py (hunk 1 above)
Comment (mislah) [reply maths/area.py:3]: Thank you @ZeroDayOwl  for you valuable suggestion.

File: maths/area.py (hunk 3 above)
Comment (mislah) [reply maths/area.py:505]: Thank you @ZeroDayOwl for your suggestion.

File: maths/area.py (hunk 4 above)
Comment (mislah) [reply maths/area.py:529]: Hi @ZeroDayOwl , Based on the format of preexisting functions, I doubt if I'm allowed to proceed with your suggestion.

File: maths/area.py (hunk 2 above)
Comment (mislah) [reply maths/area.py:458]: Hey there! doctests were already present in there. I wonder if that's a mistake.

**Review by cclauss — APPROVED** [review]

**Pull request merged**